Find a functionally equivalent code:android.view.ViewRootImpl.InputStage.deliver:COMMENT
Method Modifier: public      final       
Comment:/**
 * Delivers an event to be processed.
 */

@@ -1,15 +1,8 @@
 {
     if ((q.mFlags & QueuedInputEvent.FLAG_FINISHED) != 0) {
         forward(q);
-    } else if (mView == null || !mAdded) {
-        Slog.w(TAG, "Dropping event due to root view being removed: " + q.mEvent);
+    } else if (shouldDropInputEvent(q)) {
         finish(q, false);
-    } else if (!mAttachInfo.mHasWindowFocus && !q.mEvent.isFromSource(InputDevice.SOURCE_CLASS_POINTER) && !isTerminalInputEvent(q.mEvent)) {
-        // If this is a focused event and the window doesn't currently have input focus,
-        // then drop this event.  This could be an event that came back from the previous
-        // stage but the window has lost focus in the meantime.
-        Slog.w(TAG, "Dropping event due to no window focus: " + q.mEvent);
-        finish(q, false);
     } else {
         apply(q, onProcess(q));
     }

